### PR TITLE
ObjectWatcher.Delete checks if resource can be deleted

### PR DESCRIPTION
**Pull request by whitewindmills**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Split from https://github.com/karmada-io/karmada/pull/4813.
See https://github.com/karmada-io/karmada/pull/4813#discussion_r1560759930

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Thanks @whitewindmills 
/assign

**Comment by XiShanYongYe-Chang** [comment]
Ask @chaunceyjiang to help take a review.
/cc @chaunceyjiang

File: pkg/controllers/execution/execution_controller.go [hunk 1]
```diff
@@ -154,28 +154,6 @@ func (c *Controller) tryDeleteWorkload(clusterName string, work *workv1alpha1.Wo
...
-			klog.Errorf("Failed to get resource %v from member cluster, err is %v ", workload.GetName(), err)
-			return err
-		}
-
-		// Avoid deleting resources that not managed by karmada.
-		if util.GetLabelValue(clusterObj.GetLabels(), util.ManagedByKarmadaLabel) != util.ManagedByKarmadaLabelValue {
-			klog.Infof("Abort deleting the resource(kind=%s, %s/%s) exists in cluster %v but not managed by karmada", clusterObj.GetKind(), clusterObj.GetNamespace(), clusterObj.GetName(), clusterName)
-			return nil
-		}
-		util.MergeLabel(workload, workv1alpha2.WorkPermanentIDLabel, util.GetLabelValue(work.Labels, workv1alpha2.WorkPermanentIDLabel))
```
[the remark this answers is not in the document]
Comment (whitewindmills) [reply pkg/controllers/execution/execution_controller.go:177]: we don't need it anymore, it's never be used.

[the remark this answers is not in the document]
Comment (XiShanYongYe-Chang) [reply pkg/controllers/execution/execution_controller.go:177]: That's actually what I was wondering too. Why did it need to be added before? :see_no_evil:

[the remark this answers is not in the document]
Comment (whitewindmills) [reply pkg/controllers/execution/execution_controller.go:177]: :exploding_head: I have to admit it was my mistake.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm

File: pkg/controllers/execution/execution_controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (RainbowMango) [reply pkg/controllers/execution/execution_controller.go:177]: I was surprised with this line, that's the reason why I always ask for small PRs.

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**